admin-dev/themes/new-theme/js/components/grid/extension/async-submit-bulk-action-extension.js

```diff
@@ -0,0 +1,238 @@
+   * as argument
+   * @param {function|null} progressCallback - if defined it tracks progress of executed ajax calls. First parameter
+   * is response object.Second parameter is currently sent ids and last parameter is the total ids array
+   * @return {{extend: (function(*=): void)}}
+   */
+  constructor({
+    successCallback = null,
+    errorCallback = null,
+    progressCallback = () => {},
+  } = {}) {
+    this.progressCallback = progressCallback;
+    this.successCallback = this.setSuccessCallback(successCallback);
+    this.errorCallback = this.setErrorCallback(errorCallback);
+
+    return {
+      extend: (grid) => this.extend(grid),
+    };
+  }
+
+  setSuccessCallback(callback) {
+    return this.isCallback(callback) ? callback
+      : (response) => {
+        if ('message' in response) {
+          window.showSuccessMessage(response.message);
+        }
+
+        window.location.reload();
+      };
+  }
+
+  setErrorCallback(callback) {
+    return this.isCallback(callback) ? callback
+      : (response) => {
+        if ('message' in response) {
+          window.showErrorMessage(response.message);
+        }
+
+        window.location.reload();
+      };
+  }
+
+  isCallback(callback) {
+    return callback && typeof callback === 'function';
+  }
+
+  /**
+   * Extend grid with bulk action submitting
+   *
+   * @param {Grid} grid
+   */
+  extend(grid) {
+    grid.getContainer().on('click', '.js-bulk-action-submit-async-btn', (event) => {
+      this.submit(event, grid);
+    });
+  }
+
+  /**
+   * Collects required data and submits the form.
+   * @param event
+   * @param grid
+   */
+  submit(event, grid) {
+    const $submitBtn = $(event.currentTarget);
+    const confirmMessage = $submitBtn.data('confirm-message');
+    const confirmTitle = $submitBtn.data('confirmTitle');
+    const hasConfirmTitle = confirmTitle !== undefined;
+    const hasConfirmationMessage = confirmMessage !== undefined && confirmMessage.length > 0;
+    const hasAllConfirmationMessages = hasConfirmationMessage && hasConfirmTitle;
+
+    if (hasAllConfirmationMessages) {
+      this.showConfirmModal($submitBtn, grid, confirmMessage, confirmTitle);
+    } else if (hasConfirmationMessage && window.confirm(confirmMessage)) {
+      this.submitForm($submitBtn, grid);
+    } else {
+      this.submitForm($submitBtn, grid);
+    }
+  }
+
+  submitForm($submitBtn, grid) {
+    const $form = $(`#${grid.getId()}_filter_form`);
+    const $checkedInputs = this.getCheckedInputs($form);
+    const ids = this.getCheckedIds($checkedInputs);
+
+    const inputName = this.getInputName($checkedInputs);
+    const chunkSize = parseInt($submitBtn.data('chunkSize'), 10);
+    const chunkedIds = this.chunkArray([...ids], chunkSize);
+    const submitMethod = $submitBtn.data('form-method');
+    const formUrl = $submitBtn.data('form-url');
+
+    if (!['POST', 'GET', 'DELETE', 'PUT'].includes(submitMethod)) {
```

Comment:
Convert `submitMethod` to upper case when you retrieve it from the data, so that this extension doesn't block the behaviour in case someone uses `get`, `post`, ...